### PR TITLE
Construct a MethodDefinition node without mutating another node.

### LINKED ISSUE #1414: Consistent construction of a method definition node
Currently, a `MethodDefinition` node (see `parseClassElement`) is constructed by mutating an existing `Property` node. It is better to have a special constructor for this node (just like others in `nodes.ts`) and have the node created the same consistent way.

### PR DESCRIPTION
**Pull request by ariya**
Fixes #1414

### CODE REVIEW

File: src/parser.ts [hunk 1]
```diff
@@ -3115,14 +3119,12 @@ class Parser {
...
-        property.static = isStatic;
-        if (property.kind === 'init') {
-            property.kind = 'method';
+        if (kind === 'init') {
+            kind = 'method';
         }
         if (!isStatic) {
-            if (!property.computed && (property.key.name || property.key.value.toString()) === 'constructor') {
-                if (property.kind !== 'method' || !property.method || property.value.generator) {
+            if (!computed && this.isPropertyKey(key, 'constructor')) {
```
Comment (mikesherov) [inline src/parser.ts]: is there any test you can add that checks the `(key.type === Syntax.Literal && key.value === value)` part of `this.isProperty` for this condition?

File: src/parser.ts [hunk 2]
```diff
@@ -3130,18 +3132,15 @@ class Parser {
                 } else {
                     hasConstructor.value = true;
                 }
-                property.kind = 'constructor';
+                kind = 'constructor';
             }
         } else {
-            if (!property.computed && (property.key.name || property.key.value.toString()) === 'prototype') {
+            if (!computed && this.isPropertyKey(key, 'prototype')) {
```
Comment (mikesherov) [inline src/parser.ts]: is there any test you can add that checks the `(key.type === Syntax.Literal && key.value === value)` part of `this.isProperty` for this condition?

Comment (mikesherov) [inline src/parser.ts]: can you collapse the `else { if {` into `else if {` please?

**Comment by mikesherov** [comment]
Besides the nits... :+1:

File: src/parser.ts (hunk 2 above)
Comment (ariya) [inline src/parser.ts]: Good point. And now that I've taken a second look, I think the conditional block needs be based on `!computed` anyway for improved readability.

File: src/parser.ts (hunk 1 above)
Comment (ariya) [inline src/parser.ts]: Actually it doesn't count as a constructor and thus this case should be excluded.

**Pull request closed without merging**